Lib/asyncio/tasks.py

```diff
@@ -931,18 +931,22 @@ def callback():
 
 
 def create_eager_task_factory(custom_task_constructor):
-
-    if "eager_start" not in inspect.signature(custom_task_constructor).parameters:
-        raise TypeError(
-            "Provided constructor does not support eager task execution")
+    """
+    Creates a function suitable for use as a task factory on an event-loop.
+    
+    The tasks created will be started immediately (rather than being first
+    scheduled to an event loop). The constructor argument can be any callable
+    that returns a ``Task`` compatible object and has a signature compatible
+    with ``Task.__init__``, including the ``eager_start`` keyword argument.
+    """
```

Comment:
```suggestion
    """Create a function suitable for use as a task factory on an event-loop.

    Suggested usage:

        loop.set_task_factory(create_eager_task_factory(my_task_constructor))

    Now, tasks created will be started immediately (rather than being first
    scheduled to an event loop). The constructor argument can be any callable
    that returns a Task-compatible object and has a signature compatible
    with `Task.__init__`; it must have the `eager_start` keyword argument.
    """
```

Reply:
In addition, it should probably mention that there's already a perfectly good eager task factory for the default Task class, `asyncio.eager_task_factory`.

Reply:
```suggestion
    """Create a function suitable for use as a task factory on an event-loop.
	
	    Example usage:
	
	        loop.set_task_factory(
	            asyncio.create_eager_task_factory(my_task_constructor))
	
	    Now, tasks created will be started immediately (rather than being first
	    scheduled to an event loop). The constructor argument can be any callable
	    that returns a Task-compatible object and has a signature compatible
	    with `Task.__init__`; it must have the `eager_start` keyword argument.
	    
	    Most applications will use `Task` for `my_task_constructor` and in this
	    case there's no need to call `create_eager_task_factory()` directly.
	    Instead the  global `eager_task_factory` instance can be used. E.g.
	    `loop.set_task_factory(asyncio.eager_task_factory)`.
	    """
```